client.go

```diff
@@ -40,6 +47,13 @@ type Client struct {
 	w   *io.PipeWriter
 }
 
+type evenHandlerFunc func(context.Context, []byte) error
+
+type WSEvents struct {
```

Comment:
if this is independent of gitopia client, can u please move it to a separate file?